### PR TITLE
add public-viewer config CRD

### PR DESCRIPTION
**Pull request by filariow**
This PR adds the changes made from running `make generate` on codeready-toolchain/api after adding the `PublicViewerConfig` optional field to `ToolchainConfig`.

linked to https://github.com/codeready-toolchain/api/pull/419

### CODE REVIEW

**Review by mfrancisc — COMMENTED** [review]
Shouldn't this contain the new toolchainconfig CRD with the changes dispatched by the `make generate` command in the api repo ?

**Comment by filariow** [comment]
> Shouldn't this contain the new toolchainconfig CRD with the changes dispatched by the `make generate` command in the api repo ?

Yes, the file the `make generate` copied to this repo is `config/crd/bases/toolchain.dev.openshift.com_toolchainconfigs.yaml`. I also added the replace rule for e2e test.

**Comment by mfrancisc** [comment]
> Yes, the file the make generate copied to this repo is config/crd/bases/toolchain.dev.openshift.com_toolchainconfigs.yaml

I'm not sure why I haven't saw that in the first review, maybe some temporary issue with GH which was showing only the `go.mod` and `go.sum` changes. Sorry for the confusion.

**Review by mfrancisc — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -113,3 +113,5 @@ require (
 )
 
 go 1.20
+
+replace github.com/codeready-toolchain/api => github.com/filariow/toolchain-api v0.0.0-20240430081956-c9bf496de75d
```
Comment (alexeykazakov) [inline go.mod:117]: Let's move it to a separate PR. See the reasoning here: https://redhat-internal.slack.com/archives/C06MJ2DVBU4/p1713987983373619?thread_ts=1713955118.062449&cid=C06MJ2DVBU4

Comment (fbm3307) [reply go.mod:117]: +1

**Comment by filariow** [comment]
superseded by https://github.com/codeready-toolchain/host-operator/pull/1047
